source/val/validation_state.cpp

```diff
@@ -394,5 +394,34 @@ void ValidationState_t::RegisterInstruction(
   if (id) {
     all_definitions_.insert(make_pair(id, &ordered_instructions_.back()));
   }
+
+  // If the instruction is using an OpTypeSampledImage as an operand, it should
+  // be recorded. The validator will ensure that all usages of an
+  // OpTypeSampledImage and its definition are in the same basic block.
+  for (auto i = 0; i < inst.num_operands; ++i) {
```

Comment:
In this declaration of i "auto" is a longer fancy name for "int".  Since the literal value 0 is of type int.  :-)